extensions/query-history/src/utils.ts

```diff
@@ -11,3 +11,35 @@
 export function removeNewLines(str: string): string {
 	return str.replace(/\r\n/g, ' ').replace(/\n/g, ' ');
 }
+
+function decorate(decorator: (fn: Function, key: string) => Function): Function {
```

Comment:
I suppose you copied this from the `src/vs`, does this one do the job? https://www.npmjs.com/package/debounce-decorator-ts

Reply:
Yeah - it's originally from VS (we use it in a lot of extensions). 

For stuff like this which is only a dozen or so lines I usually prefer just to copy over the code. Packages can be problematic and often times have extra stuff we don't care about (plus make it more difficult to actually know what it's doing without spending a bunch of time digging into the source)